lib/ansible/modules/remote_management/ucs/ucs_timezone.py

```diff
@@ -0,0 +1,171 @@
+            - The timezone name is case sensitive.
+            - This name can be between 0 and 510 alphanumeric characters.
+            - You cannot use spaces or any special characters other than
+            - "\"-\" (hyphen), \"_\" (underscore), \"/\" (backslash)."
+        required: false
+
+    descr:
+        description:
+            - A user-defined description of the timezone object.
+            - Enter up to 256 characters.
+            - "You can use any characters or spaces except the following:"
+            - "` (accent mark), \ (backslash), ^ (carat), \" (double quote), = (equal sign), > (greater than), < (less than), or ' (single quote)."
+        required: no
+
+extends_documentation_fragment:
+    - ucs
+
+requirements:
+
+    - ucsmsdk
+
+author:
+
+    - John McDonough (@movinalot)
+    - CiscoUcs (@CiscoUcs)
+'''
+
+EXAMPLES = r'''
+- name: Configure Time Zone
+  ucs_timezone:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    state: present
+    admin_state: enabled
+    timezone: America/Los_Angeles
+    descr: 'Time Zone for Los Angeles'
+
+- name: Unconfigure Time Zone
+  ucs_timezone:
+    hostname: 172.16.143.150
+    username: admin
+    password: password
+    state: absent
+    admin_state: disabled
+'''
+
+RETURN = r'''
+#
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.remote_management.ucs import UCSModule, ucs_argument_spec
+
+
+def main():
+    argument_spec = ucs_argument_spec
+    argument_spec.update(
+        timezone=dict(type='str'),
+        descr=dict(type='str'),
+        admin_state=dict(type='str', default='enabled', choices=['disabled', 'enabled']),
+        state=dict(type='str', default='absent', choices=['present', 'absent']),
+    )
+
+    module = AnsibleModule(
+        argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'present', ['timezone']],
+        ],
+    )
+    ucs = UCSModule(module)
+
+    err = False
+
+    # UCSModule creation above verifies ucsmsdk is present and exits on failure, so additional imports are done below.
+
+    changed = False
+    try:
+        mo_exists = False
+        props_match = False
+
+        dn = 'sys/svc-ext/datetime-svc'
+
+        mo = ucs.login_handle.query_dn(dn)
+        if mo:
+            mo_exists = True
+
+        if module.params['state'] == 'absent':
+            # mo must exist but all properties do not have to match
+            if mo_exists:
+                if not module.check_mode:
+                    mo.timezone = ""
+                    mo.descr = ""
+                    ucs.login_handle.add_mo(mo, modify_present=True)
+                    ucs.login_handle.commit()
+                changed = True
+        else:
+            if mo_exists:
+                # check top-level mo props
+                kwargs = dict(timezone=module.params['timezone'])
+                kwargs['descr'] = module.params['descr']
```

Comment:
Based on the argument spec, does this work when no descr is provided?  I've had to default to empty string to avoid a key lookup error in module.params.

Reply:
Defaulted to empty string in argument.spec, however I had not experienced and error when it was not set.